### PR TITLE
Fixes #200: Goodbye GithubResource

### LINKED ISSUE #200: Remove GithubResource and IT tests related to it
Github authentication is no longer performed by Catapult. It is handled by Keycloak.

### PR DESCRIPTION
**Pull request by gastaldi**
- [X] Have you created an [issue](https://github.com/redhat-kontinuity/catapult/issues) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/redhat-kontinuity/catapult/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

**Comment by gastaldi** [comment]
@ALRubinger can you review it?

**Comment by gastaldi** [comment]
Worth mentioning that I also changed the `GITHUB_USERNAME` and `GITHUB_TOKEN` env vars to `CATAPULT_GITHUB_USERNAME` and `CATAPULT_GITHUB_TOKEN` respectively

**Comment by ALRubinger** [comment]
@gastaldi Would like to understand the user flow for MiniShift if we remove this; how would the user authorize access for us to push to GitHub in this case if we're not getting the token from Keycloak?

**Comment by gastaldi** [comment]
On minishift, CATAPULT_GITHUB_TOKEN env var will be used (which was already part of the process). On Openshift, Keycloak will provide the token, and this env var will not be set

**Comment by gastaldi** [comment]
So Minishift will work for a single user only. For multiple users, the CATAPULT_KEYCLOAK_URL will be used instead

**Comment by gastaldi** [comment]
Still the UI could authenticate with Keycloak, but in the backend it would just use the token defined as an env var

**Comment by gastaldi** [comment]
We should consider at some point having our own Keycloak instance, and get rid of any specific authentication code in Catapult.

**Comment by ALRubinger** [comment]
Ah, user passes `CATAPULT_GITHUB_TOKEN` when processing the template, thus granting us access.

File: README.md [hunk 1]
```diff
@@ -27,6 +27,24 @@ Prerequisites to Build
...
+```
+#!/bin/sh 
+
+export CATAPULT_GITHUB_USERNAME=<replace with your github username>
+export CATAPULT_GITHUB_TOKEN=<replace with your personal token (see step 1)>
+export CATAPULT_OPENSHIFT_API_URL=`minishift console --url`
+export CATAPULT_OPENSHIFT_CONSOLE_URL=`minishift console --url`
+export CATAPULT_KEYCLOAK_URL=http://sso.prod-preview.openshift.io
+export CATAPULT_KEYCLOAK_REALM=fabric8
+export CATAPULT_OPENSHIFT_USERNAME=admin
```
Comment (ALRubinger) [inline README.md]: developer/developer?

File: README.md [hunk 2]
```diff
@@ -35,46 +53,22 @@ Prerequisites to Run Integration Tests
...
     For instance you may put into your `~/.bash_profile`:
 
-        export GITHUB_USERNAME=ALRubinger
-        export GITHUB_TOKEN=[token created from above]
-        export GITHUB_PASSWORD=mygithub-password
+        export CATAPULT_GITHUB_USERNAME=<your github username>
+        export CATAPULT_GITHUB_TOKEN=<token created from above>
     
-    Updates to `~./bash_profile` may require logging out of the shell to be visible; you may check by typing into a terminal:
+    Use `source ~./bash_profile` to make your changes visible; you may check by typing into a terminal:
```
Comment (ALRubinger) [inline README.md]: Should now be covered by `catapult-env.sh` above, no longer in `.bash_profile`, right?

File: README.md [hunk 3]
```diff
@@ -35,46 +53,22 @@ Prerequisites to Run Integration Tests
             * `repo`
             * `admin:repo_hook`
             * `delete_repo`
-    * Create 3 environment variables:
-        * `GITHUB_USERNAME`
-        * `GITHUB_TOKEN`
-        * `GITHUB_PASSWORD`
+    * Create 2 environment variables:
```
Comment (ALRubinger) [inline README.md]: As covered above in the env script; does this need to be mentioned here?

Comment (gastaldi) [inline README.md]: I believe so. This is for the people who decided to not follow the TLDR section

File: README.md (hunk 2 above)
Comment (gastaldi) [inline README.md]: Probably yes. Will update that

File: README.md [hunk 4]
```diff
@@ -27,6 +27,24 @@ Prerequisites to Build
 
 Prerequisites to Run Integration Tests
 --------------------------------------
+TLDR; copy that to a `catapult-env.sh` file and execute it using `source catapult-env.sh`:
```
Comment (ALRubinger) [inline README.md]: Can't execute `source` on this until MiniShift is running :P

**Comment by ALRubinger** [comment]
7d167ce93fda92741eb9bed8a8ed49e936cd3d9d

**Pull request closed without merging**